app/src/main/res/layout/activity_join.xml

```diff
@@ -1,42 +1,39 @@
 <?xml version="1.0" encoding="utf-8"?>
 <RelativeLayout
     xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:custom="http://schemas.android.com/apk/res-auto"
     xmlns:tools="http://schemas.android.com/tools"
     android:id="@+id/activity_join"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
-    android:paddingBottom="@dimen/activity_vertical_margin"
-    android:paddingLeft="@dimen/activity_horizontal_margin"
-    android:paddingRight="@dimen/activity_horizontal_margin"
-    android:paddingTop="@dimen/activity_vertical_margin"
     tools:context="io.intrepid.contest.screens.entrysubmission.join.JoinActivity"
     >
 
-    <io.intrepid.contest.customviews.ClearableEditText
-        android:id="@+id/enter_code_text_view"
-        android:layout_width="match_parent"
-        android:layout_height="wrap_content"
-        android:layout_below="@id/contestant_welcome_text_view"
-        android:hint="@string/enter_code"
+    <Button
+        android:id="@+id/enter_code_submit_button"
+        style="@style/AppTheme.BottomNavigationButton"
+        android:enabled="false"
+        android:text="@string/common_submit"
         />
 
-
-    <android.support.design.widget.BottomNavigationView
-        android:id="@+id/bottom_navigation"
+    <FrameLayout
```

Comment:
you shouldnt need a frame layout here, your hint label edit text is its own layout and should be able to be inflated right in the parent.